<launch>
    <arg name="uav_id" default="2"/>

    <group ns="uav_$(arg uav_id)">
	
	<node pkg="dji_osdk_ros" type="dji_vehicle_node" name="vehicle_node" output="screen">
    	    <!-- node parameters -->
   	    <param name="acm_name" type="string" value="/dev/ttyACM0"/>
    	    <param name="serial_name" type="string" value="/dev/ttyUSB0"/>
            <param name="baud_rate" type="int" value="921600"/>
            <param name="app_id" type="int" value="1079136"/>
            <param name="app_version" type="int" value="1"/>
            <param name="align_time" type="bool" value="false"/>
            <param name="enc_key" type="string" value="073de76d2c7fd320c482e58442a62fc0ad32b18bbc6099f9c969a2473158f11e"/>
            <param name="use_broadcast" type="bool" value="false"/>
        </node>


        <node pkg="onboard_dji" type="dji_mission_node" name="mission_node" output="screen">
        <!-- node parameters -->
            <param name="damping" type="double" value="0.0"/>
            <param name="start_altitude" type="double" value="2.0"/>
            <param name="uav_id" type="double" value="$(arg uav_id)"/>
        </node>
    </group>
</launch>
